R/utilities-units.R

```diff
@@ -152,3 +152,23 @@ getDimensionTask <- function() {
   }
   return(dimensionTask)
 }
+
+
+
+#'Loop through dimensions and build a list containing an enum of all units available for each dimension
+#' @return enum of all units for each dimension
+#' @export
+getUnitsEnum <- function(){
+  dimensions <- allAvailableDimensions()
+  units <- lapply(dimensions,function(dimension){enum(getUnitsForDimension(dimension = dimension))})
+  names(units) <- dimensions
+  return(units)
+}
+
+#' #'Function to return an enum of all available dimensions
+#' @return enum of all dimensions
+#' @export
+getDimensionsEnum <- function(){
+   dimensions <- enum(allAvailableDimensions())

```

Comment:
Can you add a test that loads these to ensure that it works as expected?
Also why do we need the getEnumDimensiob?

Reply:
> Also why do we need the getEnumDimensiob?

Only because it complements getUnitsEnum and does a similar job, but we could stick with allAvailableDimensions().  But then the user would need to create the enum.  Prefer to remove it?

Reply:
oh no I see... now. You cannot use the other one directly
Sthg like x<-getUnitEnum
x$Mass

that would not work right?

I feel that we are going to load those two enums on load (see @PavelBal other PR). So I am going to merge this as soon as we have tests for this and they we can consolidate with his work

Reply:
Yes x$Mass would give the units, not "Mass"